### PR TITLE
Several fixes for scd4x.

### PR DESCRIPTION
**Pull request by fabi125**
- Fix byte packing when sending data (unsigned char instead of signed char).
- Fix condition in set_temperature_offset.
- Fix reading data in get_temperature_offset.
- Add get/set_automatic_self_calibration_enabled.
- Add persist_settings.
- Format file with isort+black.

There are currently not a lot of tests for this library. If you feel strongly about that I can add some, otherwise I've tested all my changes with the actual sensor.

### CODE REVIEW

File: library/scd4x/__init__.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import time
 import struct
+import time
+
 from smbus2 import SMBus, i2c_msg
 
-__version__ = '0.0.1'
+__version__ = "0.0.1"
```
Comment (Gadgetoid) [inline library/scd4x/__init__.py:6]: I believe this change will break the version checks in my very tenuous Makefile. Though the answer might be to fix the Makefile with something a bit less fragile.

Comment (fabi125) [reply library/scd4x/__init__.py:6]: Seems less fragile than you might think :)

```
pi@raspberrypi:~/git/scd4x-python $ grep version library/scd4x/__init__.py 
__version__ = "0.0.1"
pi@raspberrypi:~/git/scd4x-python $ make usage
Library: scd4x
Version: 0.0.1
(...)
```

**Comment by fabi125** [comment]
Up to you! I might still look into adding at least a few tests based on the example of the sensor data sheet. Just to get a little coverage.

**Comment by Gadgetoid** [comment]
I would certainly welcome tests if you're up for it!

**Comment by fabi125** [comment]
All the time I got today. Will add some tests for the remaining functions later this week.

**Comment by fabi125** [comment]
Alright, that should be all :)

**Comment by Gadgetoid** [comment]
Thank you! Sorry I've been preoccupied over the Easter hols. Slowing getting back up to speed now.

**Comment by qinvav** [comment]
Please merge it. The library is barely functional without these changes.

**Comment by riodoro1** [comment]
Come on, just merge it. Somebody else fixes the bugs for you and you can't even click a green button?

**Comment by 46bit** [comment]
@Gadgetoid Hey, could this get merged please?

**Comment by Gadgetoid** [comment]
@46bit since you asked nicely.

**Pull request merged**